aptos-move/aptos-gas/src/params.rs

```diff
@@ -1,12 +1,23 @@
 // Copyright (c) Aptos
 // SPDX-License-Identifier: Apache-2.0
 
+macro_rules! expand_get_for_gas_parameters {
+    ($params: ident . $name: ident, $map: ident, $prefix: literal, optional $key: literal) => {
+        if let Some(val) = $map.get(&format!("{}.{}", $prefix, $key)) {
+            $params.$name = (*val).into();
+        }
+    };
+    ($params: ident . $name: ident, $map: ident, $prefix: literal, $key: literal) => {
+        $params.$name = $map.get(&format!("{}.{}", $prefix, $key)).cloned()?.into();
+    };
+}
+
 macro_rules! define_gas_parameters {
     (
         $params_name: ident,
         $prefix: literal,
         [$(
-            [$name: ident: $ty: ty, $key: literal $(,)?, $initial: expr $(,)?]
+            [$name: ident: $ty: ty, $(optional $($dummy: ident)?)? $key: literal $(,)?, $initial: expr $(,)?]
```

Comment:
Another dark magic I learned today. Feeling proud and sad at the same time. Should probably just rewrite the whole thing using proc macro some day.

Reply:
Agreed, this is really too much to parse....